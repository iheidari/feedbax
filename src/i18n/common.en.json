{
  "date-format": "MM/DD/YYYY",
  "Title": "Title",
  "Description": "Description",
  "Delete": "Delete",
  "Previous": "Previous",
  "Next": "Next",
  "Save": "Save",
  "Back": "Back",
  "Ok": "OK",
  "Cancel": "Cancel",
  "Confirmation": "Confirmation",
  "Are you sure?": "Are you sure?",
  "Deleted Successfully": "Deleted Successfully",
  "Saved Successfully": "Saved Successfully",
  "None": "None"
}
