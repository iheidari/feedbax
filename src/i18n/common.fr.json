{
  "date-format": "DD-MM-YYYY",
  "Title": "Titre",
  "Description": "Description",
  "Delete": "Supprimer",
  "Previous": "Précédent",
  "Next": "Suivant",
  "Save": "Sauvegarder",
  "Back": "Retour",
  "Ok": "D'accord",
  "Cancel": "Annuler",
  "Confirmation": "Confirmation",
  "Are you sure?": "Êtes-vous sûr?",
  "Deleted Successfully": "Supprimé avec succès",
  "Saved Successfully": "Enregistré avec succès",
  "None": "Non"
}
