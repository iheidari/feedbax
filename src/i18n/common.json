{
  "en": {
    "translation": { "welcome-to-react": "Welcome to React and react-i18next" }
  },
  "fr": {
    "translation": { "welcome-to-react": "Bienvenue Ã  React et react-i18next" }
  }
}
