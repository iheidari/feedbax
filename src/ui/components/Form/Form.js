import React from 'react';
import PropTypes from 'prop-types';
import Grid from '@material-ui/core/Grid';
import { getControllerByModel } from '../../../util/ui';
import { useTranslation } from 'react-i18next';

const Form = ({ dataModel, uiModel, onModelChange }) => {
  const { t } = useTranslation();
  let formControls = [];
  for (let field in uiModel.fields) {
    const model = uiModel.fields[field];
    const dataValue = dataModel[field];
    const fieldMainControl = getControllerByModel(
      field,
      model,
      dataValue,
      onModelChange,
      t
    );
    const fieldControl = (
      <Grid key={field} item {...model.size}>
        {fieldMainControl}
      </Grid>
    );
    formControls.push(fieldControl);
  }
  return <Grid container>{formControls}</Grid>;
};

Form.propTypes = {
  dataModel: PropTypes.object.isRequired,
  uiModel: PropTypes.object.isRequired,
  onModelChange: PropTypes.func
};

export default Form;
