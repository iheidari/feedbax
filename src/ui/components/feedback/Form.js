import React from 'react';
import PropTypes from 'prop-types';
import Grid from '@material-ui/core/Grid';
import TextField from '../basic/TextField';
import DatePicker from '../basic/DatePicker';
import Switch from '../basic/Switch';
import Combobox from '../basic/Combobox';

const Form = ({ data, onModelChange }) => {
  return (
    <Grid container>
      <Grid item xs={12}>
        <TextField
          label='Title'
          margin='normal'
          variant='outlined'
          value={data.title}
          onChange={onModelChange('title')}
        />
      </Grid>
      <Grid item xs={12}>
        <TextField
          label='Description'
          margin='normal'
          variant='outlined'
          value={data.description}
          multiline
          onChange={onModelChange('description')}
        />
      </Grid>
      <Grid item xs={12}>
        <DatePicker
          label='Date'
          margin='normal'
          variant='outlined'
          value={data.date}
          onChange={onModelChange('date')}
        />
      </Grid>
      <Grid item xs={12}>
        <Switch
          label='General'
          value={data.isPublic}
          onChange={onModelChange('isPublic')}
        />
      </Grid>
      <Grid item xs={12}>
        <Combobox
          label='Category'
          value={data.category}
          hasNone
          items={[{ key: 1, value: 'one' }, { key: 2, value: 'two' }]}
          onChange={onModelChange('category')}
        />
      </Grid>
    </Grid>
  );
};

Form.propTypes = {
  data: PropTypes.object,
  onModelChange: PropTypes.func
};

export default Form;
