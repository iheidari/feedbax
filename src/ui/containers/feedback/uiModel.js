import TextField from '@material-ui/core/TextField';
import { required } from '../../../util/customValidator';

const uiModel = {
  list: {
    fields: {
      title: { text: 'Title', tooltip: '', size: { xs: 3 }, sortable: true },
      description: {
        text: 'Description',
        tooltip: '',
        size: { xs: 7 },
        sortable: false
      },
      deleteAction: { text: '', tooltip: '', size: { xs: 2 }, sortable: false }
    }
  },
  form: {
    fields: {
      title: {
        control: TextField,
        properties: {
          required: true,
          error: true,
          helperText: 'helper',
          label: 'Title',
          margin: 'normal',
          variant: 'outlined'
        },
        size: { xs: 12 },
        validation: [{ validator: required, messageId: 'isRequired' }]
      },
      description: {
        control: TextField,
        properties: {
          helperText: ' ',
          label: 'Description',
          variant: 'outlined',
          multiline: true
        },
        size: { xs: 12 },
        validation: []
      }
    }
  }
};

export default uiModel;
