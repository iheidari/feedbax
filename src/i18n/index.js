import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import commonEn from './common.en.json';
import commonFr from './common.fr.json';

const resources = {
  en: { translation: commonEn },
  fr: { translation: commonFr }
};

i18n
  .use(initReactI18next) // passes i18n down to react-i18next
  .init({
    resources,
    lng: 'en',
    fallbackLng: 'en',
    interpolation: {
      escapeValue: false
    }
  });

export default i18n;
