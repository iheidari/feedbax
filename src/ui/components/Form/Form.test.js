import React from 'react';
import Form from './Form';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import uiModel from './Form.test.mock';
import '../../../i18n';

Enzyme.configure({ adapter: new Adapter() });

describe('components/Form', () => {
  test('normal render', () => {
    const onModelChange = modelName => () => modelName;
    const dataModel = { title: 'hello' };
    const formControl = shallow(
      <Form
        dataModel={dataModel}
        uiModel={uiModel.form}
        onModelChange={onModelChange}
      />
    );
    expect(formControl).toMatchSnapshot();
  });
});
