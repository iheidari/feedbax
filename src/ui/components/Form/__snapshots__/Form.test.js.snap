// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Form normal render 1`] = `
<WithStyles(Grid)
  container={true}
>
  <WithStyles(Grid)
    item={true}
    key="title"
    xs={12}
  >
    <TextField
      0="h"
      1="e"
      2="l"
      3="l"
      4="o"
      id="title"
      label="Title"
      margin="normal"
      onChange={[Function]}
      required={false}
      select={false}
      tooltip=""
      variant="outlined"
    />
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
